lib/pliny/log.rb

```diff
@@ -4,6 +4,10 @@ def log(data, &block)
       log_to_stream(stdout || $stdout, merge_log_contexts(data), &block)
     end
 
+    def log_without_context(data, &block)
```

Comment:
@gudmundur Yeah, actually that makes perfect sense in retrospect. As a slight alternative, what would you say about a setup where you dependency injected a Pliny configuration class a little like this?

``` patch
diff --git a/lib/pliny/middleware/canonical_log_line.rb b/lib/pliny/middleware/canonical_log_line.rb
index 90be252..8083ce2 100644
--- a/lib/pliny/middleware/canonical_log_line.rb
+++ b/lib/pliny/middleware/canonical_log_line.rb
@@ -24,6 +24,13 @@ module Pliny::Middleware
       include Pliny::CanonicalLogLineHelpers
 
       #
+      # general config
+      #
+
+      log_field :app_name, String
+      log_field :deployment, String
+
+      #
       # error
       #
 
@@ -56,8 +63,14 @@ module Pliny::Middleware
       log_field :timing_total_elapsed, Float
     end
 
-    def initialize(app, emitter:)
+    def initialize(app, config:, emitter:)
       @app = app
+
+      # Note that we dependency inject "config" because a global `Config`
+      # symbol is only available in the context of an actual Pliny app, and not
+      # within the test suite of Pliny itself.
+      @config = config
+
       @emitter = emitter
     end
 
@@ -70,6 +83,15 @@ module Pliny::Middleware
           line = LogLine.new
 
           #
+          # general config
+          #
+
+          if @config
+            line.app_name = @config.app_name
+            line.deployment = @config.deployment
+          end
+
+          #
           # error
           #
 
diff --git a/lib/template/lib/routes.rb b/lib/template/lib/routes.rb
index 6a48bed..531e806 100644
--- a/lib/template/lib/routes.rb
+++ b/lib/template/lib/routes.rb
@@ -5,6 +5,7 @@ Routes = Rack::Builder.new do
   use Pliny::Middleware::RequestStore::Seed, store: Pliny::RequestStore
   use Pliny::Middleware::Metrics
   use Pliny::Middleware::CanonicalLogLine,
+      config: Config,
       emitter: -> (data) {
         Pliny.log_without_context({ canonical_log_line: true }.merge(data))
       }
diff --git a/spec/middleware/canonical_log_line_spec.rb b/spec/middleware/canonical_log_line_spec.rb
index 51888d9..efab16a 100644
--- a/spec/middleware/canonical_log_line_spec.rb
+++ b/spec/middleware/canonical_log_line_spec.rb
@@ -1,12 +1,23 @@
 require "spec_helper"
 
 describe Pliny::Middleware::CanonicalLogLine do
+  class TestConfig
+    def app_name
+      "my-app"
+    end
+
+    def deployment
+      "production"
+    end
+  end
+
   def app
     Rack::Builder.new do
       run Sinatra.new {
         use Pliny::Middleware::RequestID
 
         use Pliny::Middleware::CanonicalLogLine,
+          config: TestConfig.new,
           emitter: -> (data) { Pliny.log_without_context(data) }
 
         use Pliny::Middleware::RescueErrors, raise: false
@@ -34,6 +45,10 @@ describe Pliny::Middleware::CanonicalLogLine do
     header "User-Agent", "rack-test"
     get "/apps/123"
 
+    config = TestConfig.new
+    assert_equal config.app_name, data[:app_name]
+    assert_equal config.deployment, data[:deployment]
+
     assert_match Pliny::Middleware::RequestID::UUID_PATTERN,
       data[:request_id]
     assert_equal "127.0.0.1", data[:request_ip]
```

I sort of like it because all the fields to be included in the canonical line stay very explicit. The downside is that once again it makes customization somewhat difficult until we start vendoring the middleware into new Pliny apps. Thoughts?